frontend/app/routes/admin.js

```diff
@@ -0,0 +1,13 @@
+import Route from '@ember/routing/route';
+import Ember from 'ember';
+
+const { isEmpty, inject } = Ember;
+
+export default Route.extend({
+  session: inject.service(),
+  beforeModel(transition) {
+    if (isEmpty(this.get('session').get('token'))) {
```

Comment:
How can you check for the user or admin from here as  every authenticated user will have token